src/Disks/IDisk.h

```diff
@@ -404,6 +404,11 @@ class IDisk : public Space
     virtual bool supportsChmod() const { return false; }
     virtual void chmod(const String & /*path*/, mode_t /*mode*/) { throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Disk does not support chmod"); }
 
+    /// Was disk created to be used without storage configuration?
+    bool isCustomDisk() const { return is_custom_disk; }
```

Comment:
Yes, because there is code that does manipulations with disks based on what is inside `context->getDisksMap` (which gets disks from `DiskSelector`). For example, `MergeTreeData::loadDataParts` (where custom disks must also be checked):
https://github.com/ClickHouse/ClickHouse/blob/8fde6067684fb2af626feb4b2be03f5c15d6f7f4/src/Storages/MergeTree/MergeTreeData.cpp#L1274

Moreover, (first, please see a comment below https://github.com/ClickHouse/ClickHouse/pull/41976#discussion_r1063517891)
in this PR we have not only custom disks but also a feature to specify disk name instead of storage policy,
so we now can unify them both without adding a separate code for them if we put custom disk in disk selector:
https://github.com/ClickHouse/ClickHouse/blob/8fde6067684fb2af626feb4b2be03f5c15d6f7f4/src/Storages/MergeTree/MergeTreeData.cpp#L382-L385